### PR TITLE
Make lazy-loading the default for next/image

### PR DESCRIPTION
**Pull request by timneutkens**
Also fixed a flexbox rendering issue that @samsisle reported while testing.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: ef79ffc17dbf610b0a651024cc52a468a0884f70

**test/integration/image-component/basic/test/index.test.js** 
- Image Component Tests > Client-side Image Component Tests > should support the unoptimized attribute
- Image Component Tests > SSR Image Component Tests > should support the unoptimized attribute

<details>
<summary>Expand output</summary>

  ● Image Component Tests › SSR Image Component Tests › should support the unoptimized attribute

    expect(received).toBe(expected) // Object.is equality

    Expected: "https://arbitraryurl.com/foo.jpg"
    Received: null

      59 |     expect(
      60 |       await browser.elementById('unoptimized-image').getAttribute('src')
    > 61 |     ).toBe('https://arbitraryurl.com/foo.jpg')
         |       ^
      62 |   })
      63 |   it('should not add a srcset if unoptimized attribute present', async () => {
      64 |     expect(

      at Object.<anonymous> (integration/image-component/basic/test/index.test.js:61:7)

  ● Image Component Tests › Client-side Image Component Tests › should support the unoptimized attribute

    expect(received).toBe(expected) // Object.is equality

    Expected: "https://arbitraryurl.com/foo.jpg"
    Received: null

      59 |     expect(
      60 |       await browser.elementById('unoptimized-image').getAttribute('src')
    > 61 |     ).toBe('https://arbitraryurl.com/foo.jpg')
         |       ^
      62 |   })
      63 |   it('should not add a srcset if unoptimized attribute present', async () => {
      64 |     expect(

      at Object.<anonymous> (integration/image-component/basic/test/index.test.js:61:7)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.3s | -124ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.45 | 2.375 | -0.08 |
| / avg req/sec | 1020.2 | 1052.81 | +32.61 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.331 | 1.314 | -0.02 |
| /error-in-render avg req/sec | 1878.23 | 1902.31 | +24.08 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 994 B | 994 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| buildDuration | 14.2s | 13.8s | -403ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +281 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _error.js | 1.06 MB | 1.06 MB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.06 MB | 1.06 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.1 MB | 1.1 MB | ✓ |
| withRouter.js | 1.1 MB | 1.1 MB | ✓ |
| Overall change | 5.42 MB | 5.42 MB | ✓ |

</details>

</details>
Commit: ef79ffc17dbf610b0a651024cc52a468a0884f70

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| buildDuration | 12.7s | 13s | ⚠️ +241ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +637 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.389 | 2.36 | -0.03 |
| / avg req/sec | 1046.54 | 1059.28 | +12.74 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.394 | 1.311 | -0.08 |
| /error-in-render avg req/sec | 1793.38 | 1907.23 | +113.85 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 994 B | 994 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| buildDuration | 14.5s | 14.7s | ⚠️ +292ms |
| nodeModulesSize | 91.2 MB | 91.2 MB | ⚠️ +637 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..35e7.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-23d7afb..be6f.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ccae7c3..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/image-component-loading | Change |
| - | - | - | - |
| _error.js | 1.06 MB | 1.06 MB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.06 MB | 1.06 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.1 MB | 1.1 MB | ✓ |
| withRouter.js | 1.1 MB | 1.1 MB | ✓ |
| Overall change | 5.42 MB | 5.42 MB | ✓ |

</details>

</details>
Commit: 6c11e4fadcb277ea4272fa123552bc59851850e2

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 6c11e4fadcb277ea4272fa123552bc59851850e2

**test/integration/image-component/basic/test/index.test.js** 
- Image Component Tests > Client-side Image Component Tests > should support the unoptimized attribute
- Image Component Tests > SSR Image Component Tests > should support the unoptimized attribute

<details>
<summary>Expand output</summary>

  ● Image Component Tests › SSR Image Component Tests › should support the unoptimized attribute

    expect(received).toBe(expected) // Object.is equality

    Expected: "https://arbitraryurl.com/foo.jpg"
    Received: null

      59 |     expect(
      60 |       await browser.elementById('unoptimized-image').getAttribute('src')
    > 61 |     ).toBe('https://arbitraryurl.com/foo.jpg')
         |       ^
      62 |   })
      63 |   it('should not add a srcset if unoptimized attribute present', async () => {
      64 |     expect(

      at Object.<anonymous> (integration/image-component/basic/test/index.test.js:61:7)

  ● Image Component Tests › Client-side Image Component Tests › should support the unoptimized attribute

    expect(received).toBe(expected) // Object.is equality

    Expected: "https://arbitraryurl.com/foo.jpg"
    Received: null

      59 |     expect(
      60 |       await browser.elementById('unoptimized-image').getAttribute('src')
    > 61 |     ).toBe('https://arbitraryurl.com/foo.jpg')
         |       ^
      62 |   })
      63 |   it('should not add a srcset if unoptimized attribute present', async () => {
      64 |     expect(

      at Object.<anonymous> (integration/image-component/basic/test/index.test.js:61:7)

</details>

**Review by leerob — DISMISSED** [review]

**Review by styfle — DISMISSED** [review]
Nice catch 👍

File: packages/next/client/image.tsx [hunk 1]
```diff
@@ -156,11 +156,14 @@ export default function Image({
   // If priority and lazy are present, log an error and use priority only.
   if (priority && lazy) {
     if (process.env.NODE_ENV !== 'production') {
-      console.error(
-        `Image with src ${src} has both priority and lazy tags. Only one should be used.`
+      throw new Error(
```
Comment (styfle) [inline packages/next/client/image.tsx:159]: I noticed you throw instead of log error. Do we also want to throw on line 255?

Comment (timneutkens) [reply packages/next/client/image.tsx:159]: Yep!

Comment (styfle) [reply packages/next/client/image.tsx:159]: Done!

**Pull request merged**